components/script/dom/htmlinputelement.rs

```diff
@@ -292,12 +320,32 @@ impl HTMLInputElementMethods for HTMLInputElement {
 
     // https://html.spec.whatwg.org/multipage/#dom-input-value
     fn Value(&self) -> DOMString {
-        self.textinput.borrow().get_content()
+        match self.get_value_mode() {
+            ValueMode::Filename | ValueMode::Value => self.textinput.borrow().get_content(),
```

Comment:
This is actually much more involved than just returning the `textinput`.The spec says the following for `value`:
> On getting, it must return the current value of the element. On setting, it must set the element's value to the new value, set the element's dirty value flag to true, invoke the value sanitization algorithm, if the element's type attribute's current state defines one, and then, if the element has a text entry cursor position, should move the text entry cursor position to the end of the text field, unselecting any selected text and resetting the selection direction to none.

And the following for `filename`:
> On getting, it must return the string "C:\fakepath\" followed by the name of the first file in the list of selected files, if any, or the empty string if the list is empty. On setting, if the new value is the empty string, it must empty the list of selected files; otherwise, it must throw an InvalidStateError exception.